vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -305,45 +344,20 @@ type internal FSharpProjectOptionsManager
     /// Get the exact options for a document or project
     member this.TryGetOptionsForDocumentOrProject(document: Document) =
         async { 
-            let projectId = document.Project.Id
-
-            // The options for a single-file script project are re-requested each time the file is analyzed.  This is because the
-            // single-file project may contain #load and #r references which are changing as the user edits, and we may need to re-analyze
-            // to determine the latest settings.  FCS keeps a cache to help ensure these are up-to-date.
-            match singleFileProjectTable.TryGetValue(projectId) with
-            | true, (loadTime, _, _) ->
-                try
-                    let fileName = document.FilePath
-                    let! cancellationToken = Async.CancellationToken
-                    let! sourceText = document.GetTextAsync(cancellationToken) |> Async.AwaitTask
-                    // NOTE: we don't use FCS cross-project references from scripts to projects.  The projects must have been
-                    // compiled and #r will refer to files on disk.
-                    let tryGetOrCreateProjectId _ = None 
-                    let! _referencedProjectFileNames, parsingOptions, projectOptions = this.ComputeSingleFileOptions (tryGetOrCreateProjectId, fileName, loadTime, sourceText.ToString(), document.Project.Solution)
-                    this.AddOrUpdateSingleFileProject(projectId, (loadTime, parsingOptions, projectOptions))
-                    return Some (parsingOptions, None, projectOptions)
-                with ex -> 
-                    Assert.Exception(ex)
-                    return None
+            match! reactor.TryGetOptionsByDocumentAsync(document) with
+            | Some(parsingOptions, projectOptions) ->
+                return Some(parsingOptions, None, projectOptions)
             | _ ->
-                match! reactor.TryGetOptionsByProjectAsync(document.Project) with
-                | Some(parsingOptions, projectOptions) ->
-                    return Some(parsingOptions, None, projectOptions)
-                | _ ->
-                    return None
+                return None
         }
 
     /// Get the options for a document or project relevant for syntax processing.
     /// Quicker then TryGetOptionsForDocumentOrProject as it doesn't need to recompute the exact project options for a script.
     member this.TryGetOptionsForEditingDocumentOrProject(document:Document) = 
-        let projectId = document.Project.Id
```

Comment:
@TIHan I don't understand this change in dev16.  

https://github.com/dotnet/fsharp/issues/10156 is a synchronous call to this on the UI thread

As the comment to this method says, the point of "TryGetOptionsForEditingDocumentOrProject" is to get editing options quickly for scripts, in the cases where we need them fast on the UI thread for editing operations.   But this change removed this and now this method computed the exact options.

I believe this is a primary cause for slower script editing in dev16, e.g. https://github.com/dotnet/fsharp/issues/6646, and is related to the UI thread aspects of https://github.com/dotnet/fsharp/issues/8757

Reply:
Looking back, I don't think I understood this well enough at the time and made the change in hopes to normalize everything.

If this call was being invoked on the UI thread somewhere, I can totally understand why there would be a performance issue.